sdk/storage/azure-storage-queue/azure/storage/queue/_models.py

```diff
@@ -27,19 +37,29 @@ class QueueAnalyticsLogging(GeneratedLogging):
     :keyword bool delete: Required. Indicates whether all delete requests should be logged.
     :keyword bool read: Required. Indicates whether all read requests should be logged.
     :keyword bool write: Required. Indicates whether all write requests should be logged.
-    :keyword ~azure.storage.queue.RetentionPolicy retention_policy: Required.
-        The retention policy for the metrics.
+    :keyword ~azure.storage.queue.RetentionPolicy retention_policy: The retention policy for the metrics.
     """
 
-    def __init__(self, **kwargs):
+    version: str = '1.0'
+    """The version of Storage Analytics to configure."""
+    delete: bool = False
+    """Indicates whether all delete requests should be logged."""
+    read: bool = False
+    """Indicates whether all read requests should be logged"""
+    write: bool = False
+    """Indicates whether all write requests should be logged."""
+    retention_policy: "RetentionPolicy"
+    """The retention policy for the metrics."""
```

Comment:
Just want to point this out that CI was not happy with `= RetentionPolicy()` so going to leave as no default.